modules/extensions-spring-support/src/test/java/org/mule/module/extension/internal/DynamicConfigExpirationTestCase.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.extension.internal;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+import org.mule.api.MuleEvent;
+import org.mule.module.extension.HeisenbergExtension;
+import org.mule.tck.junit4.ExtensionsFunctionalTestCase;
+import org.mule.tck.probe.JUnitProbe;
+import org.mule.tck.probe.PollingProber;
+
+import org.junit.Test;
+
+public class DynamicConfigExpirationTestCase extends ExtensionsFunctionalTestCase
+{
+
+    @Override
+    protected String getConfigFile()
+    {
+        return "dynamic-config-expiration.xml";
+    }
+
+    @Override
+    protected Class<?>[] getAnnotatedExtensionClasses()
+    {
+        return new Class<?>[] {HeisenbergExtension.class};
+    }
+
+    @Test
+    public void expireDynamicConfig() throws Exception
+    {
+        final String myName = "Walt";
+        final MuleEvent event = getTestEvent(myName);
+        String returnedName = runFlow("dynamic", event).getMessage().getPayloadAsString();
+
+
+        HeisenbergExtension config = muleContext.getRegistry().lookupObject(HeisenbergExtension.class);
+
+        // validate we actually hit the correct dynamic config
+        assertThat(returnedName, is(myName));
+        assertThat(config.getPersonalInfo().getName(), is(myName));
+
```

Comment:
Why do you need to use "whatever TimeSupplier the platform uses"? You are not testing the time supplier